### DISCUSSION TITLE
Please leave us feedback!! We would LOVE to hear from you!!

### DISCUSSION THREAD

**Discussion by rarajase** [microsoftgraph/group-membership-management#9] [General]
Hello!!

Have you had a chance to download and try out the GMM code in this open source?

The GMM tool makes it easy to manage a large group roster by taking advantage of existing security groups and/or smaller groups kept up to date by teams within the larger org. The tool essentially takes security groups or other Microsoft 365 Groups as source group(s), produces an aggregated flat-list membership roster of the destination Microsoft 365 Group based on the source groups and continues to keep the destination group roster in sync with the source groups.

We are standing by to hear from you!! We would LOVE to hear feedback on what worked, what challenges you are facing or any other input you have! Please drop us a note on  this discussion forum or send us an email at GMMSupport@service.microsoft.com.

Looking forward to hearing back from you,
Regards,
Group Membership Management team

**Comment by sellarsm1** [comment]
When an Office 365 group is set to dynamic, the membership management UI is suppressed in Teams.  Could a similar flag be introduced for scenarios like this so that Teams owners don't get confused or frustrated when changes they make don't 'stick'?  
Alternatively, if there were more multi-value attributes available in AzureAD, a routine like this could simply add tags that could drive membership in a dynamic group.
Thanks, Martin

**Reply by rarajase** [reply]
Hi, Thank you for your comment Martin! Can you please provide a little more detail around your experience with GMM, your company, use case scenarios? Were you able to successfully deploy and use this tool? We agree closing the membership management in application end points is a feature that would be greatly beneficial. But, not having an attribute in AAD to leverage for this has been slowing this down. The team will continue to explore options, but, one thing we have done is ask group owners update the title to indicate that the membership is being managed by admin and to reach out to group owners with changes.

**Comment by sellarsm1** [comment]
Hi, to be Frank, I haven’t tried the GMM tool yet, I simply read about it in the O365 CAB forum.  In our environment, we have many (>1,000) groups whose membership is managed programmatically based on attributes, hierarchy , etc and we have a mechanism to explicitly include ad hoc members that might not otherwise match the filter.   These groups are synced to AzureAD, EXO and SPO and used as distribution lists, security groups or both. As our adoption of Teams expands, I’m confident some Teams owners will want to base their membership on these groups.

If we could establish a multi valued attribute (either directly in AAD or as an extension attribute) we could use our existing group management mechanisms and simply have them set tags that could be used in O365 groups dynamic rules to mange the Teams membership). This approach would have the added benefit of suppressing membership management in the Teams UI.

Martin

________________________________

**Comment by gcmlpfaltaher** [comment]
This repo doesn't seem to be maintained by MS, are there plans to update this repo or roadmap plans to introduce this sort of functionality as a feature released by MS?

**Reply by DG-RMD** [reply]
Would very much love to hear an answer to this, the solution has many caveats to overcome in its current state

**Comment by bkinnan2** [comment]
Is the repo and code still being actively maintained and updated? It looks like there have been no contributions to it since last year.

If not, is there a new program or offering coming out to fill this gap?

**Comment by alrios-ms** [comment]
Hi @bkinnan2, it is actively maintained, we are working on a major change for the installation process as per the feedback we got it was a very complex process, we are expecting to update the repo with a new installation process and code very soon.
